web_embedding/cra-flutter/src/App/App.tsx

```diff
@@ -0,0 +1,217 @@
+import IconButton from '@mui/material/IconButton'
+import MenuIcon from '@mui/icons-material/Menu'
+import FlutterDashIcon from '@mui/icons-material/FlutterDash'
+import Drawer from '@mui/material/Drawer'
+import CssBaseline from '@mui/material/CssBaseline'
+import List from '@mui/material/List'
+import FormControl from '@mui/material/FormControl'
+import MenuItem from '@mui/material/MenuItem'
+import InputLabel from '@mui/material/InputLabel'
+import Select, { SelectChangeEvent } from '@mui/material/Select'
+import TextField from '@mui/material/TextField'
+import { FlutterView } from './FlutterView/FlutterView'
+
+const drawerWidth = 300
+
+const Main = styled('main', { shouldForwardProp: (prop) => prop !== 'open' })<{
+  open?: boolean;
+}>(({ theme, open }) => ({
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  flexGrow: 1,
+  transition: theme.transitions.create('margin', {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  marginLeft: `-${drawerWidth}px`,
+  ...(open && {
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginLeft: 0,
+  }),
+}))
+
+const DrawerHeader = styled('div')(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  padding: theme.spacing(0, 1),
+  // necessary for content to be below app bar
+  ...theme.mixins.toolbar,
+  justifyContent: 'flex-end',
+}))
+
+const FlutterAppWrapper = styled('div')(({ theme }) => ({
+  border: '1px solid #eee',
+  borderRadius: '5px',
+  height: '480px',
+  width: '320px',
+  transition: theme.transitions.create('all', {
+    easing: theme.transitions.easing.easeInOut,
+    duration: theme.transitions.duration.enteringScreen,
+  }),
+  overflow: 'hidden',
+}))
+
+const EffectButton = ({ title, toggleClassName }: { title: string, toggleClassName: (name: string) => void }) => {
+  const onClick = () => {
+    toggleClassName(`fx-${title.toLowerCase()}`)
+  }
+  return (
+    <Button
+      variant="outlined"
+      onClick={onClick}
+      sx={{ minWidth: '130px' }}
+    >
+      {title}
+    </Button>
+  )
+}
+
+function App () {
+  const theme = useTheme()
+  const [drawerOpened, setDrawerOpened] = React.useState(false)
+  const [classNames, setClassNames] = React.useState('')
+  const [screen, setScreen] = React.useState('counter')
+  const [clicks, setClicks] = React.useState(0)
+  const [text, setText] = React.useState('')
+  const flutterState = useRef<any>(null)
+  const handleDrawer = () => {
+    setDrawerOpened(!drawerOpened)
+  }
+  const onFlutterAppLoaded = (state: any) => {
+    flutterState.current = state
+    state.onClicksChanged(() => { setClicks(flutterState.current?.getClicks()) })
+    state.onTextChanged(() => { setText(flutterState.current?.getText()) })
+    state.onScreenChanged(() => { setScreen(flutterState.current?.getScreen()) })
+  }
```

Comment:
@p-mazhnik from a ReactJS standpoint, how weird is this interaction? Is there anything we could change in the state object coming from flutter so this is more natural for a ReactJS site?

Reply:
I would say that a more natural flow is when we receive an updated value from the subscription,
so here we can have 
```js
state.onScreenChanged((screen) => { setScreen(screen) })
```

Reply:
Makes sense, this should be a simple change in the `class DemoAppStateManager`. It could also expose the values as proper `getters`/`setters` rather than named `get/setSomething` methods, so `state.text` is a thing that can be read from and written to. (I wonder if `@JSExport` supports that!)